src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -637,38 +632,39 @@ internal MemberAccessor MemberAccessorStrategy
             }
         }
 
-        /// <summary>
-        /// JsonTypeInfo resolver used for serialization.
-        /// </summary>
-        internal IJsonTypeInfoResolver? _typeInfoResolver;
+        internal bool IsInitializedForReflectionSerializer { get; private set; }
 
         /// <summary>
         /// Initializes the converters for the reflection-based serializer.
         /// </summary>
         [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
         [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
         [MemberNotNull(nameof(_typeInfoResolver))]
-        internal void EnsureInitializedForReflectionSerializer()
+        internal void InitializeForReflectionSerializer()
```

Comment:
Why not keep it with `EnsureInitialized` -  that removes couple of repeating `if`s

Reply:
Since it's being called by the hot path, it's best if we avoid calling this moderately sized method.